packages/cloudflare/src/index.ts

```diff
@@ -88,6 +90,50 @@ export default function createIntegration(args?: Options): AstroIntegration {
 								disabled: !args?.wasmModuleImports,
 							}),
 							chunkAnalyzer.getPlugin(),
+							{
+								name: 'prerender-scanner',
+								enforce: 'post',
+								generateBundle(_, bundle) {
+									for (const chunk of Object.values(bundle)) {
+										if (chunk.type !== 'chunk') continue;
+										if (chunk.dynamicImports.some((entry) => entry.includes('prerender'))) {
+											prerenderImports.push(chunk.fileName);
+										}
+									}
+								},
+							},
+							{
+								name: 'prerender-worker',
+								enforce: 'post',
+								generateBundle(_, bundle) {
+									for (const chunk of Object.values(bundle)) {
+										if (chunk.type !== 'chunk') continue;
+										if (chunk.name === '_@astrojs-ssr-virtual-entry') {
+											chunk.dynamicImports = chunk.dynamicImports.filter(
+												(entry) => !prerenderImports.includes(entry)
+											);
+											for (const page of prerenderImports) {
+												const importRegex = new RegExp(
+													`^const (_page\\d) = \\(\\) => import\\('.\\/${page}'\\);$\\n`,
+													'gm'
+												);
+												let pageId: string | undefined;
+												const matches = chunk.code.matchAll(importRegex);
+												for (const match of matches) {
+													if (match[1]) {
+														pageId = match[1];
+													}
+												}
+												chunk.code = chunk.code.replace(importRegex, '');
+												if (pageId) {
+													const arrayRegex = new RegExp(`^^ +\\[".+", ${pageId}\\]$\\n`, 'gm');
```

Comment:
[Regex diagram](https://regexper.com/#%2F%5E%5E%20%2B%5C%5B%22.%2B%22%2C%20_page123%5C%5D%24%5Cn%2Fgm)

![image](https://github.com/withastro/adapters/assets/11063910/d31e8307-f846-46a6-92eb-a1244c20a5f6)

I don't know what is the code that this Regex is trying to match, but it is looking for:
- The start of an array
- The start of a string
- Anything
- The end of a string
- The value of the variable named the value `pageId`
- The end of a line
- A following line
- Is entirely contained within a single line

And it doesn't need to match the start of the line twice (`^^` could be `^`). [Some examples](regexr.com/7ukvv).

Is that the expected pattern?

Reply:
@Fryuni the goal is to match and remove from the `pageMap`, e.g.

```diff
const pageMap = new Map([
    ["../../../../../node_modules/.pnpm/astro@4.5.8_@types+node@18.17.8_typescript@5.2.2/node_modules/astro/dist/assets/endpoint/generic.js", _page0],
    ["src/pages/add/[a]/[b].ts", _page1],
-    ["src/pages/hybrid.ts", _page2]
]);
```

However we don't get `src/pages/hybrid.ts` in Vite IIUC, so the only think we know for sure to match is `_page2`. Any ideas how to either get the source filename or make the regex safer?

Reply:
That information can be retrieved from the module graph, on `astro` it is done by [these helper functions](https://github.com/withastro/astro/blob/0fec72b35cccf80b66a85664877ca9dcc94114aa/packages/astro/src/prerender/metadata.ts).

Reply:
@Fryuni How can I retrieve the initial source path e.g. `src/pages/hybrid.ts` inside `generateBundle`.
The helper functions astro core uses, don't work inside the adapter 🤔